tests/test_query_params.py

```diff
@@ -64,6 +64,14 @@ def test_percent_encoded(self):
         self.assertEqual(req.get_param_as_list('id', int), [23, 42])
         self.assertEqual(req.get_param('q'), u'\u8c46 \u74e3')
 
+    def test_bad_percentage(self):
+        query_string = 'x=%%20%+%'
```

Comment:
Please also test mixing in some non-escaped text (e.g., 'peregrine'), as well as some additional non-hex-pair, percent-escaped characters (e.g., '%a', '%z', '%zz').